File: eSalon/eSalon.Services/Database/Recenzija.cs
using System;
using System.Collections.Generic;

namespace eSalon.Services.Database;

public partial class Recenzija : ISoftDelete
{
    public int RecenzijaId { get; set; }

    public int KorisnikId { get; set; }

    public int UslugaId { get; set; }

    public string Komentar { get; set; } = null!;

    public DateTime DatumDodavanja { get; set; }

    public int BrojLajkova { get; set; }

    public int BrojDislajkova { get; set; }

    public bool IsDeleted { get; set; }

    public DateTime? VrijemeBrisanja { get; set; }

    public virtual Korisnik Korisnik { get; set; } = null!;

    public virtual ICollection<RecenzijaOdgovor> RecenzijaOdgovors { get; set; } = new List<RecenzijaOdgovor>();

    public virtual ICollection<RecenzijaReakcija> RecenzijaReakcijas { get; set; } = new List<RecenzijaReakcija>();

    public virtual Usluga Usluga { get; set; } = null!;
}

File: eSalon/UI/esalon_desktop/lib/models/recenzija.dart
import 'package:json_annotation/json_annotation.dart';

part 'recenzija.g.dart';

@JsonSerializable()
class Recenzija {
  int? recenzijaId;
  int korisnikId;
  int uslugaId;
  String komentar;
  DateTime datumDodavanja;
  int brojLajkova;
  int brojDislajkova;
  String? korisnickoIme;
  String? nazivUsluge;

  Recenzija(
    this.recenzijaId,
    this.korisnikId,
    this.uslugaId,
    this.komentar,
    this.datumDodavanja,
    this.brojLajkova,
    this.brojDislajkova,
    this.korisnickoIme,
    this.nazivUsluge,
  );
  
  factory Recenzija.fromJson(Map<String, dynamic> json) => _$RecenzijaFromJson(json);

  Map<String, dynamic> toJson() => _$RecenzijaToJson(this);
}